### DISCUSSION TITLE
PRs are created ahead of specified `stabilityDays`

### DISCUSSION THREAD

**Discussion by mcmxcdev** [renovatebot/renovate#18499] [Request Help] [answered]
We are new to managing dependencies with renovate and are wondering if `stabilityDays` config works properly.

We only care about major version updates and want to delay all updates by 14 days.
Now we woke up to new PRs (`schedule:weekly`), even though `nx` v15 has only been released 10 days ago.
We would've expected `nx` v15 to pop up next week, since then it would be 17d, and it would qualify to be part of the `schedule:weekly` check.

Below you can see our `renovate.json`. Is there something wrong with our config or does renovate somehow prioritize `schedule:weekly` over `stabilityDays` ?

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "schedule:weekly",
    ":preserveSemverRanges",
    "helpers:disableTypesNodeMajor"
  ],
  "labels": ["dependencies"],
  "enabledManagers": ["npm", "github-actions"],
  "ignoreDeps": ["googleapis", "react-phone-number-input"],
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "enabled": false,
      "stabilityDays": 14,
      "internalChecksFilter": "strict"
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "enabled": false,
      "matchPackagePatterns": ["mantine"],
      "groupName": "mantine"
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "enabled": false,
      "matchPackagePatterns": ["temporalio"],
      "groupName": "temporalio"
    }
  ]
}

```

Created PR:

![Screenshot from 2022-10-24 09-45-12](https://user-images.githubusercontent.com/16797721/197482834-294417c0-9e12-4410-a55c-9e304c6ee3da.png)

**Comment by RahulGautamSingh** [comment, marked as answer]
Hey sorry for the delayed response,

You will need to add a new packageRule which matches the major update type and put stability info in it. Also the stability info in the first packageRule won't be necessary as you have already disbaled them.
Example:
```
   {
      "matchUpdateTypes": ["major"],
      "stabilityDays": 14
    }
```

**Comment by viceice** [comment]
I don't think `stabilityDays` works with `updateTypes` when used with `"internalChecksFilter": "strict"` because the filtering is then done before we know the updateType.

@rarkins Are I'm right?

**Reply by rarkins** [reply]
Yes, filtering is doing early when `internalChecksFilter` is `strict`.
